supervisor_checks/utils.py

```diff
@@ -87,3 +89,155 @@ def get_port(port_or_port_re, process_name):
     raise errors.InvalidCheckConfig(
         'Could not extract port number for process name %s using regular '
         'expression %s' % (process_name, port_or_port_re))
+
+
+
+
+# Directly copied from the tempfile module
+class _TemporaryFileCloser:
```

Comment:
If these two are just copy and paste from stdlib, maybe we should just import them:

```
In [1]: from tempfile import _TemporaryFileCloser
In [2]: from tempfile import _TemporaryFileWrapper
```

I understand importing protected stuff is not optimal, still looks better than carrying a bunch of stuff from standard library here.

Reply:
In general importing already written code looks better to me too. I was just worried about the potential removal or change of these two internal classes in future python versions. As they are not public classes, this could be the case, despite being unlikely perhaps. Then this code would break. Wdyt?

Reply:
Ideally, I would say we ensure it doesn't break by adding proper unit tests. I don't ask you to add tests though, I think we will find out if something is broken during next Python upgrade.

Reply:
Unit tests are a great idea, but I still find it better to include these classes in this repo, with the argumentation what we would do in case the tempfile implementation in python changes. Then we would nevertheless have to write some custom code to fit our use case. These two classes are self-contained and we currently don’t depend on any other module.

Reply:
Ok, current package license is MIT. If we add some code under PSF license, does it require any disclaimer or license update? This is the main thing I am concerned about.